src/distributions/float.rs

```diff
@@ -9,176 +9,90 @@
 // except according to those terms.
 
 //! Basic floating-point number distributions
-
-
-/// A distribution to sample floating point numbers uniformly in the open
-/// interval `(0, 1)` (not including either endpoint).
-///
-/// See also: [`Closed01`] for the closed `[0, 1]`; [`Uniform`] for the
-/// half-open `[0, 1)`.
-///
-/// # Example
-/// ```rust
-/// use rand::{weak_rng, Rng};
-/// use rand::distributions::Open01;
-///
-/// let val: f32 = weak_rng().sample(Open01);
-/// println!("f32 from (0,1): {}", val);
-/// ```
-///
-/// [`Uniform`]: struct.Uniform.html
-/// [`Closed01`]: struct.Closed01.html
-#[derive(Clone, Copy, Debug)]
-pub struct Open01;
-
-/// A distribution to sample floating point numbers uniformly in the closed
-/// interval `[0, 1]` (including both endpoints).
-///
-/// See also: [`Open01`] for the open `(0, 1)`; [`Uniform`] for the half-open
-/// `[0, 1)`.
-///
-/// # Example
-/// ```rust
-/// use rand::{weak_rng, Rng};
-/// use rand::distributions::Closed01;
-///
-/// let val: f32 = weak_rng().sample(Closed01);
-/// println!("f32 from [0,1]: {}", val);
-/// ```
-///
-/// [`Uniform`]: struct.Uniform.html
-/// [`Open01`]: struct.Open01.html
-#[derive(Clone, Copy, Debug)]
-pub struct Closed01;
-
+use Rng;
+use distributions::{Distribution, Uniform};
+use ::core::mem;
+
+pub(crate) trait IntoFloat {
+    type F;
+
+    /// Helper method to combine the fraction and a contant exponent into a
+    /// float.
+    ///
+    /// Only the least significant bits of `self` may be set, 23 for `f32` and
+    /// 52 for `f64`.
+    /// The resulting value will fall in a range that depends on the exponent.
+    /// As an example the range with exponent 0 will be
+    /// [2<sup>0</sup>..2<sup>1</sup>-1), which is [1..2).
```

Comment:
2.pow(1) - 1 = 1, not 2 — or am I not reading it right?

Reply:
Oops, `-1` is not supposed to be there